tests/rptest/scale_tests/large_messages_test.py

```diff
@@ -0,0 +1,488 @@
+from ducktape.utils.util import wait_until
+
+from rptest.clients.rpk import RpkTool
+from rptest.services.cluster import cluster
+from rptest.services.admin import Admin
+from rptest.services.redpanda import RESTART_LOG_ALLOW_LIST, LoggingConfig, \
+    MetricsEndpoint, PandaproxyConfig, SchemaRegistryConfig
+from rptest.services.producer_swarm import ProducerSwarm
+from rptest.services.consumer_swarm import ConsumerSwarm
+from rptest.tests.redpanda_test import RedpandaTest
+from rptest.utils.scale_parameters import ScaleParameters
+
+
+class LargeMessagesTest(RedpandaTest):
+    # Max time to wait for the cluster to be healthy once more.
+    HEALTHY_WAIT_SECONDS = 20 * 60
+
+    # Up to 5 min to stop the node with a lot of topics
+    STOP_TIMEOUT = 60 * 5
+
+    # Progress wait timeout
+    PROGRESS_TIMEOUT = 60 * 3
+
+    LEADER_BALANCER_PERIOD_MS = 30000
+
+    def __init__(self, *args, **kwargs):
+        # Topics
+        # Prepare RP
+        super().__init__(
+            *args,
+            num_brokers=3,
+            extra_rp_conf={
+                # Disable leader balancer initially, to enable us to check for
+                # stable leadership during initial elections and post-restart
+                # elections.  We will switch it on later, to exercise it during
+                # the traffic stress test.
+                # 'enable_leader_balancer': False,
+
+                # Avoid waiting for 5 minutes for leader balancer to activate
+                "leader_balancer_idle_timeout": self.LEADER_BALANCER_PERIOD_MS,
+                "leader_balancer_mute_timeout": self.LEADER_BALANCER_PERIOD_MS,
+
+                # Tweak storage related params
+                'reclaim_batch_cache_min_free': 256000000,
+                'storage_read_buffer_size': 32768,
+                'storage_read_readahead_count': 2,
+                'disable_metrics': True,
+                'disable_public_metrics': False,
+                'append_chunk_size': 32768,
+                'kafka_rpc_server_tcp_recv_buf': 131072,
+                'kafka_rpc_server_tcp_send_buf': 131072,
+                'kafka_rpc_server_stream_recv_buf': 32768,
+
+                # Tuning
+                'kafka_batch_max_bytes': 10485760 * 2,
+
+                # Enable all the rate limiting things we would have in
+                # production, to ensure their effect is accounted for,
+                # but with high enough limits that we do
+                # not expect to hit them.
+                'kafka_connection_rate_limit': 10000,
+                'kafka_connections_max': 50000,
+
+                # In testing tiered storage, we care about creating as many
+                # cloud segments as possible. To that end, bounding the segment
+                # size isn't productive.
+                'cloud_storage_segment_size_min': 1,
+                'log_segment_size_min': 1024,
+
+                # Disable segment merging: when we create many small segments
```

Comment:
We are trying to be tougher here. RP will "grow" segment sizes to message size each time on each message. This will give even more load on the RP while processing incoming traffic.